### DISCUSSION TITLE
What should I put for the amount of extraction/download/multi-part thread count?

### DISCUSSION THREAD

**Discussion by Proutbedaine** [MalloyDelacroix/DownloaderForReddit#283] [Q&A] [answered]
I have a new PC with a i9-12900KF (24 cores [8 Performance cores that Hyperthreads into 16 + 8 Efficiency cores] 3.2GHz/5.1GHz Turbo) and I want to get the most out of it with all my apps. And I saw those settings before when using the app in my previous PC and saw it was set to 4 on all of them. I assumed it was because I had a i7-4790K without thinking much of it. But it's the same on this PC so it is there by default I guess.

Now I wondered what would be the optimal amount of threads I should put on each of those core settings? (extraction, download, multi-part). Should it be the same as the number of logical cores (24) ? Or physical cores (16) ? Or something else entirely?

**Comment by MalloyDelacroix** [comment, marked as answer]
Honestly, there's no right answer.  I would suggest you tune the settings to what works best for you.  For instance, I have a similar processor and I run with 20 extraction threads and 30 download threads.  The numbers don't necessarily have to be tied to your CPU count.  Your internet connection speed will also play a part in the optimal setting.

I would, however, suggest that you keep the extraction thread count lower than the download thread count.  You are going to be able to download more simultaneously than you will be able to extract.

It's come to my attention that there may be some issues with multi-part download.  So I would suggest not using it for the moment until I can resolve the issues with it.

**Reply by Proutbedaine** [reply]
> It's come to my attention that there may be some issues with multi-part download. So I would suggest not using it for the moment until I can resolve the issues with it.

Ah oki thanks for the reply and info. Also noted on that last one, I will deactivate it for now. :)
